### PR TITLE
Fix whitespace issue in lists

### PR DESCRIPTION
**Pull request by marcusklaas**
We would previously close lists when a line wasn't completely empty but didn't have the required indentation. Now lines that only contain whitespace are also correctly identified as empty.

### CODE REVIEW

File: tests/suite/regression.rs [hunk 1]
```diff
@@ -850,8 +850,44 @@ fn regression_test_61() {
 
 #[test]
 fn regression_test_62() {
-    let original = "<foo";
```
Comment (marcusklaas) [inline tests/suite/regression.rs:853]: Where did this testcase go???

Comment (raphlinus) [reply tests/suite/regression.rs:853]: It was added as part of #438 and should be retained, I think.

Comment (marcusklaas) [reply tests/suite/regression.rs:853]: Agreed, but it was somehow only removed from the *generated* tests. It looks like it was never added to the test source?

Comment (marcusklaas) [reply tests/suite/regression.rs:853]: Ah yes, looking at your PR that seems to be the case. I shall add it to the test source in this PR.

Comment (raphlinus) [reply tests/suite/regression.rs:853]: Right, that's what happened, I added it to the wrong file. Thanks!

File: specs/regression.txt [hunk 2]
```diff
@@ -747,3 +747,54 @@ ISSUE 421
...
+```````````````````````````````` example
+* A list.
+
+   * A sublist.
+
+   * Another sublist.
+
+
+* A list.
+
```
Comment (marcusklaas) [inline specs/regression.txt:762]: It's unfortunate that GitHub doesn't show whitespace on blank lines here. It makes this test look a little silly ^^

**Pull request merged**